logzio-log-sender/src/test/java/io/logz/BaseSenderTest.java

```diff
@@ -0,0 +1,120 @@
+package io.logz;
+
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.BindException;
+import java.net.ServerSocket;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseSenderTest {
+
+    private final static Logger logger = LoggerFactory.getLogger(BaseSenderTest.class);
+    protected static final String LISTENER_ADDRESS = "localhost";
+    protected MockLogzioBulkListener mockListener;
+    protected int port;
+    protected TestLogzioSender testLogzioSender;
+
+    @Before
+    public void startMockListener() throws Exception {
+        int attempts = 1;
+        while (attempts <= 3) {
+            int availablePort = -1;
+            try {
+                ServerSocket serverSocket = new ServerSocket(0);
+                serverSocket.close();
+                availablePort = serverSocket.getLocalPort();;
+                mockListener = new MockLogzioBulkListener(LISTENER_ADDRESS, availablePort);
+                logger.info("Starting Mock listener on port {}", availablePort);
+                mockListener.start();
+                port = availablePort;
+                break;
+            } catch (BindException e) {
+                if (attempts++ == 3) {
+                    throw new RuntimeException("Failed to get a non busy port: "+availablePort, e);
+                } else {
+                    logger.info("Failed to start mock listener on port {}", availablePort);
+                }
+            }
+        }
+    }
+
+    @After
+    public void stopMockListener() {
+        logger.info("Stopping Mock listener");
+        mockListener.stop();
+    }
+
+
+    protected TestLogzioSender createLogzioSender(String token, String type, String loggerName, Integer drainTimeout,
+                                        Integer fsPercentThreshold, String bufferDir, Integer socketTimeout) {
+
+        logger.info("Creating logger {}. token={}, type={}, drainTimeout={}, fsPercentThreshold={}, bufferDir={}, socketTimeout={},",
+                loggerName, token, type, drainTimeout, fsPercentThreshold, bufferDir, socketTimeout);
+
+        if (bufferDir == null) {
+            File tempDir = TestEnvironment.createTempDirectory();
+            tempDir.deleteOnExit();
+            bufferDir = tempDir.getAbsolutePath();
+        }
+        testLogzioSender = new TestLogzioSender(token,type,drainTimeout,fsPercentThreshold,bufferDir,
+                "http://" + LISTENER_ADDRESS + ":" + port,socketTimeout,socketTimeout,true, 30);
+
+        return testLogzioSender;
+    }
+
+    protected String random(int numberOfChars) {
+        return UUID.randomUUID().toString().substring(0, numberOfChars-1);
+    }
+
+    protected void assertNumberOfReceivedMsgs(int count) {
```

Comment:
All of this asserts can be either move to its own assertions class or move them to the mock listener class.